app/Traits/UserTrait.php

```diff
@@ -68,20 +75,24 @@ public function getUserRoleDetailsAttribute()
 
     public function getAuthorizedSolutionsAttribute()
     {
-        $solutions = collect(null);
-        switch ($this->user_role) {
-            case 'financer':
-                if ($this->partners->count()) {
-                    $solutions = $this->partners->first()->financesSolutions;
-                }
-                break;
-
-            case 'evaluator':
-                $solutions = $this->solutions;
-                break;
+        if ('applicant' === $this->user_role) {
+            return $this->applicationSubmissions;
+        } else {
+            $solutions = collect(null);
+            switch ($this->user_role) {
+                case 'financer':
+                    if ($this->partners->count()) {
+                        $solutions = $this->partners->first()->financesSolutions;
+                    }
+                    break;
+
+                case 'evaluator':
+                    $solutions = $this->solutions;
+                    break;
+            }
+
+            return $solutions->whereIn('status', config('dashboard.show_solutions_with_status'));
```

Comment:
My main issue with this function is that it's less clear what it does, now that it returns both solutions and submissions. Not sure on what to rename it yet.

Otherwise, we can turn the applicant case into a guard clause and revert the rest of the function changes.

```suggestion
        if ('applicant' === $this->user_role) {
            return $this->applicationSubmissions;
        }

        $solutions = collect(null);
        switch ($this->user_role) {
            case 'financer':
                if ($this->partners->count()) {
                    $solutions = $this->partners->first()->financesSolutions;
                }
                break;

            case 'evaluator':
                $solutions = $this->solutions;
                break;
        }

        return $solutions->whereIn('status', config('dashboard.show_solutions_with_status'));
```